tests/tests/test_inventory.py

```diff
@@ -13,39 +13,70 @@
 #    limitations under the License.
 
 import json
+import tempfile
 import time
 
 import pytest
 
 from .. import conftest
 from ..common_setup import standard_setup_one_client_bootstrapped
-from ..MenderAPI import devauth, inv, logger
+from ..MenderAPI import deploy, devauth, inv, logger
+from .common_artifact import get_script_artifact
 from .mendertesting import MenderTesting
 
 
+def make_script_artifact(artifact_name, device_type, output_path):
+    script = b"""\
+#!/bin/bash
+exit 0
+"""
+    return get_script_artifact(script, artifact_name, device_type, output_path)
+
+
 @pytest.mark.usefixtures("standard_setup_one_client_bootstrapped")
 class TestInventory(MenderTesting):
     @MenderTesting.fast
     def test_inventory(self):
-        """Test that device reports inventory after having bootstrapped."""
-
-        attempts = 10
-
-        while True:
-            attempts = attempts - 1
+        """
+        Test that device reports inventory after having bootstrapped and performed
+        an application update using a dummy script artifact.
+        """
+
+        # create a simple artifact (script) which doesn't do anything
+        artifact_name = "simple-artifact"
+        with tempfile.NamedTemporaryFile() as tf:
+            artifact = make_script_artifact(
+                artifact_name,
+                conftest.machine_name,
+                tf.name,
+                extra_args="--software-name swname --software-version v1",
```

Comment:
I can't see that this is checked anywhere. Did you forget to actually modify the pass condition after you made the bootstrapping installation? :-)

Reply:
I've added the check